### PR TITLE
CORS-3483: Update CAPI and CAPZ versions to set Machine DisableExtensionOperations

### PR DESCRIPTION
**Pull request by sadasu**
Updating CAPI to version v1.7.0 and CAPZ to v1.15.1-0.20240617212811-a52056dfb88c

The CAPZ version bump bring in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4792

And set DisableExtensionOperations to true.

Co-authored-by: Aditya Narayanaswamy <anarayan@redhat.com>

### CODE REVIEW

**Comment by sadasu** [comment]
/cc @jhixson74 , @patrickdillon, @rna-afk 

I have rebased @rna-afk 's PR https://github.com/openshift/installer/pull/8488 here.

I am still keeping https://github.com/openshift/installer/pull/8594 separate. Please let me know if I have roll it into this.

**Review by r4f4 — CHANGES_REQUESTED** [review]
This is somehow changing CAPA and breaking the build:
```
 pkg/asset/machines/aws/awsmachines.go:119:21: awsMachine.Spec.ElasticIPPool undefined (type "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2".AWSMachineSpec has no field or method ElasticIPPool)
pkg/asset/machines/aws/awsmachines.go:119:43: undefined: capa.ElasticIPPool
pkg/asset/machines/aws/awsmachines.go:121:47: undefined: capa.PublicIpv4PoolFallbackOrderAmazonPool
# github.com/openshift/installer/pkg/asset/machines/aws
# [github.com/openshift/installer/pkg/asset/machines/aws]
vet: pkg/asset/machines/aws/awsmachines.go:119:21: awsMachine.Spec.ElasticIPPool undefined (type v1beta2.AWSMachineSpec has no field or method ElasticIPPool) 
```

**Comment by r4f4** [comment]
Also capz is failing to build:
```
 go build -gcflags "" -ldflags "-s -w" -o ../../bin/linux_amd64/cluster-api-provider-azure "$path";
# k8s.io/apiserver/pkg/server/routes
vendor/k8s.io/apiserver/pkg/server/routes/openapi.go:69:89: cannot use oa.V3Config (variable of type *common.OpenAPIV3Config) as *common.Config value in argument to builder3.BuildOpenAPISpecFromRoutes 
```

File: cluster-api/providers/azure/go.mod
Comment (r4f4) [inline cluster-api/providers/azure/go.mod:142]: To be safe, this should never be higher than the version of capi we're building (1.7.0 as of today).

Comment (patrickdillon) [reply cluster-api/providers/azure/go.mod:142]: Interesting. And I see @sadasu is fixing by bumping the CAPI controller in 4b4dd048dafc9fd204cb50fc758fbfc7d213cbf0 :+1:

Comment (sadasu) [reply cluster-api/providers/azure/go.mod:142]: Updating CAPI version everywhere else too. (see later commits)

Comment (r4f4) [reply cluster-api/providers/azure/go.mod:142]: If CAPZ uses a new feature in 1.7.3, then running it against an older capi could cause runtime failures. I think what you're doing now is the correct approach. Let's just make sure to trigger jobs for all other providers and make sure they work with capi 1.7.3.

**Comment by r4f4** [comment]
`ibmcloud-capi-ovn` is expected to fail, I forgot about it.

**Comment by r4f4** [comment]
`e2e-aws-ovn`, `e2e-nutanix-capi-ovn`, `e2e-openstack-capi-ovn`, `e2e-vsphere-capi-ovn`: cluster installed, so no problems with the capi bump.

**Comment by r4f4** [comment]
`e2e-azure-capi-ovn`: I see the following repeating error line in the logs:
```
time="2024-06-18T18:52:42Z" level=debug msg="E0618 18:52:42.254498    8173 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
```
It'd be good to double check if it's a concern.

**Comment by r4f4** [comment]
It's a LGTM if the changes are not negatively impacting azure.

**Comment by patrickdillon** [comment]
looking at [the alintfra-e2e-azure-capi-ovn job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8627/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1803121823341088768#1:build-log.txt%3A254) infrastructure is not becoming ready. 

I would have expected this to fail at waiting for machines, although looking at the job history, some [previous jobs](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8126/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1783507245829459968) were actually able to complete installs. So seems like something is off.

**Comment by sadasu** [comment]
> looking at [the alintfra-e2e-azure-capi-ovn job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8627/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1803121823341088768#1:build-log.txt%3A254) infrastructure is not becoming ready.
> 
> I would have expected this to fail at waiting for machines, although looking at the job history, some [previous jobs](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8126/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1783507245829459968) were actually able to complete installs. So seems like something is off in this PR to fail at this stage.

Looking at the failure seen in this PR in isolation (not comparing it with previous failures), I see this:
```
level=info msg=Waiting up to 15m0s (until 6:33PM UTC) for network infrastructure to become ready...
level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: infrastructure was not ready within 15m0s: client rate limiter Wait returned an error: context deadline exceeded
level=info msg=Shutting down local Cluster API control plane...
```
So, this error is generated here: https://github.com/kubernetes/client-go/commit/147848c452865ee870eafa8eba223849c40e791c
Probable cause: during networking infrastructure provisioning, API calls to Azure are being rate limited and that is causing the 15min timer to run out

**Comment by jhixson74** [comment]
> `e2e-azure-capi-ovn`: I see the following repeating error line in the logs:
> 
> ```
> time="2024-06-18T18:52:42Z" level=debug msg="E0618 18:52:42.254498    8173 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
> ```
> 
> It'd be good to double check if it's a concern.

Yup. This is definitely an issue. I spent some time debugging it but am not familiar enough with cluster-api to get to the bottom of it at this time. It looks like things aren't caching correctly or in time. I decided to try and pinpoint when the changes took place that broke this and determined that to be between v1.6.6 and v1.7.0.  Unless there is a need to go to v1.7.0 right now, it would be nice to get this in at v1.6.6 and debug this later when we have more time ;-)

**Comment by r4f4** [comment]
> > `e2e-azure-capi-ovn`: I see the following repeating error line in the logs:
> > ```
> > time="2024-06-18T18:52:42Z" level=debug msg="E0618 18:52:42.254498    8173 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It'd be good to double check if it's a concern.
> 
> Yup. This is definitely an issue. I spent some time debugging it but am not familiar enough with cluster-api to get to the bottom of it at this time. It looks like things aren't caching correctly or in time. I decided to try and pinpoint when the changes took place that broke this and determined that to be between v1.6.6 and v1.7.0. Unless there is a need to go to v1.7.0 right now, it would be nice to get this in at v1.6.6 and debug this later when we have more time ;-)

Looking at [this azure run](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8283/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1801255083866853376/artifacts/e2e-azure-capi-ovn/ipi-install-install/artifacts/.openshift_install-1718296265.log) from [this PR](https://github.com/openshift/installer/pull/8283) I don't see that error line even though it's using [capi 1.7.0](https://github.com/openshift/installer/blob/master/cluster-api/cluster-api/go.mod#L5). I would expect the issue to be between `1.7.0` and `1.7.3` _if_ it's a capi problem. It must be some capi + capz combination.

From what I can tell, the error happens when running the azure controllers, specifically the `AzureMachinePool`:
```
time="2024-06-13T15:11:39Z" level=debug msg="I0613 15:11:39.333862    9018 controller.go:186] \"Starting Controller\" controller=\"azuremachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePool\""
time="2024-06-13T15:11:39Z" level=debug msg="I0613 15:11:39.335979    9018 reflector.go:351] Caches populated for *v1.Secret from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105"
time="2024-06-13T15:11:39Z" level=debug msg="I0613 15:11:39.338057    9018 reflector.go:351] Caches populated for *v1beta1.Cluster from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105"
time="2024-06-13T15:11:39Z" level=debug msg="I0613 15:11:39.338558    9018 reflector.go:351] Caches populated for *v1beta1.MachinePool from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105"
time="2024-06-13T15:11:39Z" level=debug msg="E0613 15:11:39.338605    9018 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
```
That controller only runs if the [machinePool feature gate](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/main.go#L418) is `true` which was changed to [true in capi v1.7.0](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.7.0).

The worrisome part about this log is:
```
time="2024-06-13T15:13:39Z" level=debug msg="E0613 15:13:39.336109    9018 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
time="2024-06-13T15:13:39Z" level=debug msg="E0613 15:13:39.437976    9018 controller.go:203] \"Could not wait for Cache to sync\" err=\"failed to wait for azuremachinepool caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.KubeadmConfig\" controller=\"azuremachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePool\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438049    9018 internal.go:516] \"Stopping and waiting for non leader election runnables\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438067    9018 internal.go:520] \"Stopping and waiting for leader election runnables\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438096    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePool\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438105    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azurecluster\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438112    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438121    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"ASOSecret\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438133    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438159    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremanagedmachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedMachinePool\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438170    9018 controller.go:242] \"All workers finished\" controller=\"azuremachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePool\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438174    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremachinetemplate\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachineTemplate\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438184    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremanagedcontrolplane\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedControlPlane\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438195    9018 controller.go:242] \"All workers finished\" controller=\"azuremanagedcontrolplane\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedControlPlane\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438197    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremachinepoolmachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePoolMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438208    9018 controller.go:242] \"All workers finished\" controller=\"azuremachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438215    9018 controller.go:242] \"All workers finished\" controller=\"azuremachinepoolmachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachinePoolMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438216    9018 controller.go:240] \"Shutdown signal received, waiting for all workers to finish\" controller=\"azuremanagedcluster\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438224    9018 controller.go:242] \"All workers finished\" controller=\"azuremachine\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachine\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438226    9018 controller.go:242] \"All workers finished\" controller=\"azurecluster\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438230    9018 controller.go:242] \"All workers finished\" controller=\"azuremanagedcluster\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438232    9018 controller.go:242] \"All workers finished\" controller=\"azuremachinetemplate\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureMachineTemplate\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438245    9018 controller.go:242] \"All workers finished\" controller=\"azuremanagedmachinepool\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureManagedMachinePool\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438248    9018 controller.go:242] \"All workers finished\" controller=\"ASOSecret\" controllerGroup=\"infrastructure.cluster.x-k8s.io\" controllerKind=\"AzureCluster\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438285    9018 internal.go:528] \"Stopping and waiting for caches\""
time="2024-06-13T15:13:39Z" level=debug msg="W0613 15:13:39.438375    9018 reflector.go:462] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105: watch of *v1api20220701.NatGateway ended with: an error on the server (\"unable to decode an event from the watch stream: context canceled\") has prevented the request from succeeding"
time="2024-06-13T15:13:39Z" level=debug msg="W0613 15:13:39.438501    9018 reflector.go:462] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105: watch of *v1beta1.Machine ended with: an error on the server (\"unable to decode an event from the watch stream: context canceled\") has prevented the request from succeeding"
time="2024-06-13T15:13:39Z" level=debug msg="W0613 15:13:39.438527    9018 reflector.go:462] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers.go:105: watch of *v1api20201101.VirtualNetworksSubnet ended with: an error on the server (\"unable to decode an event from the watch stream: context canceled\") has prevented the request from succeeding"
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438636    9018 internal.go:532] \"Stopping and waiting for webhooks\""
time="2024-06-13T15:13:39Z" level=debug msg="I0613 15:13:39.438682    9018 server.go:249] \"Shutting down webhook server with timeout of 1 minute\" logger=\"controller-runtime.webhook\""
time="2024-06-13T15:13:40Z" level=debug msg="I0613 15:13:40.493073    9018 internal.go:535] \"Stopping and waiting for HTTP servers\""
time="2024-06-13T15:13:40Z" level=debug msg="I0613 15:13:40.493121    9018 server.go:43] \"shutting down server\" kind=\"health probe\" addr=\"127.0.0.1:35841\""
time="2024-06-13T15:13:40Z" level=debug msg="I0613 15:13:40.493174    9018 internal.go:539] \"Wait completed, proceeding to shutdown the manager\""
time="2024-06-13T15:13:40Z" level=debug msg="E0613 15:13:40.493208    9018 main.go:353] \"problem running manager\" err=\"failed to wait for azuremachinepool caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.KubeadmConfig\" logger=\"setup\""
time="2024-06-13T15:26:49Z" level=debug msg="Collecting applied cluster api manifests..."
```
So azure controllers are shutting down (via a context cancel?) and nothing progresses for 13min until the Installer shuts envtest down at the 15min network infra timeout .

**Comment by patrickdillon** [comment]
> `e2e-azure-capi-ovn`: I see the following repeating error line in the logs:
> 
> ```
> time="2024-06-18T18:52:42Z" level=debug msg="E0618 18:52:42.254498    8173 kind.go:63] \"if kind is a CRD, it should be installed before calling Start\" err=\"no matches for kind \\\"KubeadmConfig\\\" in version \\\"bootstrap.cluster.x-k8s.io/v1beta1\\\"\" logger=\"controller-runtime.source.EventHandler\" kind=\"KubeadmConfig.bootstrap.cluster.x-k8s.io\""
> ```

`KubeadmConfig` is defined in sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1, which is not currently added to the scheme in the local control plane. I was able to resolve this issue by adding it to the scheme like this

```diff
diff --git a/pkg/clusterapi/localcontrolplane.go b/pkg/clusterapi/localcontrolplane.go
index d23ca4767a..19bd54aad8 100644
--- a/pkg/clusterapi/localcontrolplane.go
+++ b/pkg/clusterapi/localcontrolplane.go
@@ -23,6 +23,7 @@ import (
        capov1 "sigs.k8s.io/cluster-api-provider-openstack/api/v1beta1"
        capvv1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
        clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+       kubeadmv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
        "sigs.k8s.io/controller-runtime/pkg/client"
        "sigs.k8s.io/controller-runtime/pkg/envtest"
        "sigs.k8s.io/controller-runtime/pkg/log"
@@ -38,6 +39,7 @@ var (
 func init() {
        utilruntime.Must(clusterv1.AddToScheme(Scheme))
        utilruntime.Must(capav1beta1.AddToScheme(Scheme))
+       utilruntime.Must(kubeadmv1.AddToScheme(Scheme))
        utilruntime.Must(capav1.AddToScheme(Scheme))
        utilruntime.Must(capzv1.AddToScheme(Scheme))
        utilruntime.Must(capgv1.AddToScheme(Scheme))

```

**Comment by sadasu** [comment]
Thanks @patrickdillon. I have added the above change as a commit. I am still unclear about how this issue showed us in azure and not other platforms.

**Comment by patrickdillon** [comment]
>  I am still unclear about how this issue showed up in azure and not other platforms.

I think it's what Rafael said at the beginning of this https://github.com/openshift/installer/pull/8627#issuecomment-2178061050: CAPZ added some machinepool functionality in a recent version which depends on the kubadmconfig CRD,. The error we're seeing is that the CRD is not defined.

We need to follow up on Rafael's point that the MachinePool feature gate is now enabled. Disabling that may perhaps be a better way of resolving this problem. 

I don't think there are any practical issues with adding the kubeadm api to the scheme, but we're not really using it.

**Comment by patrickdillon** [comment]
> We need to follow up on Rafael's point that the MachinePool feature gate is now enabled. Disabling that may perhaps be a better way of resolving this problem.

Created https://issues.redhat.com/browse/CORS-3562 for this

**Comment by sadasu** [comment]
> > We need to follow up on Rafael's point that the MachinePool feature gate is now enabled. Disabling that may perhaps be a better way of resolving this problem.
> 
> Created https://issues.redhat.com/browse/CORS-3562 for this

Opened https://github.com/openshift/installer/pull/8637 for this.

**Comment by patrickdillon** [comment]
The kubeadmconfig error still occurred in the most recent run: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8627/pull-ci-openshift-installer-master-altinfra-e2e-azure-capi-ovn/1803807915287515136

I think my local testing did not incorporate the changes from this PR correctly, and my local testing was flawed.

https://github.com/openshift/installer/pull/8637 solves this in a better way and is in the merge pool. so we can retest once that merges.

We should probably drop e577876aa3e3dee1a4f452be20bc488595dad069, which was added on my suggestion, unless we determine it has some value

**Comment by sadasu** [comment]
Removed commit that added KubeadmConfig as a scheme in the local control plane. And https://github.com/openshift/installer/pull/8637 merged. Testing again.

**Comment by patrickdillon** [comment]
I just tested this locally and it works!

```
DEBUG Time elapsed per stage:                      
DEBUG        Infrastructure Pre-provisioning: 1s   
DEBUG    Network-infrastructure Provisioning: 4m11s 
DEBUG Post-network, pre-machine Provisioning: 19m14s 
DEBUG        Bootstrap Ignition Provisioning: 1s   
DEBUG                   Machine Provisioning: 7m16s 
DEBUG       Infrastructure Post-provisioning: 23s  
DEBUG                     Bootstrap Complete: 15m16s 
DEBUG                                    API: 20s  
DEBUG                      Bootstrap Destroy: 56s  
DEBUG            Cluster Operators Available: 12m11s 
DEBUG               Cluster Operators Stable: 47s  
INFO Time elapsed: 1h0m30s        
```


Let's try to get this in as soon as possible, because it does not look fun to rebase this...

/approve

**Comment by patrickdillon** [comment]
```
[36mINFO[0m[2024-06-21T15:15:35Z] Running step e2e-azure-capi-ovn-ipi-install-install. 
[36mINFO[0m[2024-06-21T16:21:53Z] Step e2e-azure-capi-ovn-ipi-install-install succeeded after 1h6m18s. 
```
azure install has succeeded. just waiting for e2e tests.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by jhixson74** [comment]
> I just tested this locally and it works!
> 
> ```
> DEBUG Time elapsed per stage:                      
> DEBUG        Infrastructure Pre-provisioning: 1s   
> DEBUG    Network-infrastructure Provisioning: 4m11s 
> DEBUG Post-network, pre-machine Provisioning: 19m14s 
> DEBUG        Bootstrap Ignition Provisioning: 1s   
> DEBUG                   Machine Provisioning: 7m16s 
> DEBUG       Infrastructure Post-provisioning: 23s  
> DEBUG                     Bootstrap Complete: 15m16s 
> DEBUG                                    API: 20s  
> DEBUG                      Bootstrap Destroy: 56s  
> DEBUG            Cluster Operators Available: 12m11s 
> DEBUG               Cluster Operators Stable: 47s  
> INFO Time elapsed: 1h0m30s        
> ```
> 
> Let's try to get this in as soon as possible, because it does not look fun to rebase this...
> 
> /approve

Confirmed here as well.

**Pull request merged**
